### PR TITLE
Modified get_atmosphere and get_density when single float is input fo…

### PR DESCRIPTION
**Pull request by fschlueter**
…r speed.

The present functions utilizing np.where (which works for single floats and arrays) are quite inefficient in case of an float. Since these function are in the 'inner loop' of most calculations and mostly getting floats there is a benefit in having this duplicated code pieces.
For get_distance_xmax_geometrical I got a boost of an factor of 3 in computing time.

This commit somewhat increases complexity but I gives you a speed boost of a factor of ~3 (for the numerical calculations).
I changed the np.shape(ele) == () to not hasattr(ele, "__len__"). Not sure if both are completely equivalent and if that matters here. The latter one is a bit faster.

### CODE REVIEW

**Comment by fschlueter** [comment]
I know it increases the complexity and is somewhat duplicated code. However there is a benefit once you aim for numerical integration over a curved atmosphere. One can write the functions for floats such that they can take arrays as well but I realized that takes away most of the speed advantage..

File: radiotools/atmosphere/models.py [hunk 1]
```diff
@@ -141,7 +141,8 @@ def get_vertical_height(slant_depth, model=default_model):
 
 
 def _get_vertical_height(at, model=default_model):
-    if np.shape(at) == ():
+    """ get vertical height from atmosphere, i.e., mass overburden """
+    if not hasattr(at, "__len__"):
```
Comment (cg-laser) [inline radiotools/atmosphere/models.py:145]: The difference is, I think, that `np.shape` only work if `at` is a numpy array? So checking for the existence of `__len__` seems better to me.

**Review by cg-laser — APPROVED** [review]
changes looks good. I agree that a factor 3 in speed is worth a slightly more complex code. 
I assume that you tested that the results don't change.

**Comment by cg-laser** [comment]
@fschlueter I think these changes can be merged?

**Comment by fschlueter** [comment]
Uh seems I did not catch your approval..

**Pull request merged**